view-creator-framework/build.gradle.kts

```diff
@@ -53,6 +54,7 @@ dependencies {
     exclude("org.yaml", "snakeyaml")
     exclude("org.apache.hive", "hive-storage-api")
     exclude("org.apache.datasketches", "datasketches-java")
+    exclude("io.airlift", "aircompressor")
```

Comment:
Are we excluding it or upgrading it? It would be one or the other, and we almost never want to exclude - I'm not sure what happened in this repo, looks like someone historically didn't have a complete grasp on dependency management.

The reason is exclude basically is saying - "Yes, I know this third party (or one of its dependencies) declared it needed this jar at runtime, but I'm saying they're wrong". Yes, you might be right based on the methods we call - but anything can change. We can call something different, we can upgrade and a third party can trigger a new code path etc. So the strictly safer alternative is a dependency constraint. It's not completely safe - it's still modifying the classpath of the third party. But it's going to be safer than an exclude every time since if they do call it, odds are the third party has maintained compatibility (vs it being completely missing). The only time you'd use an exclude would be if the vulnerable dependency has no upgrade available _and_ we can trust it's not being used.

Reply:
This comment still applies - we have both an exclude and a constraint currently which is incorrect